pinot-query-runtime/src/main/java/org/apache/pinot/query/runtime/executor/SchedulerService.java

```diff
@@ -0,0 +1,34 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.query.runtime.executor;
+
+import org.apache.pinot.query.runtime.operator.OpChain;
+import org.apache.pinot.query.runtime.operator.OpChainId;
+
+
+public interface SchedulerService {
+
+  void register(OpChain operatorChain);
```

Comment:
better interface would

submit(requestId, opChainId, opchain)

cancel(requestId)

notify(opChainId)

}
I dont know what awaitdataAvailable is intended to do

Reply:
for that the awaitDataAvaiable can be simply called `await(opChainId, timeoutMs)`

Reply:
await is basically yield right? the fact that we wait for data to be available is irrelevant to the scheduler..

Reply:
For the context, Java uses the terms `wait` and `notify` here. I've found a bit confusing the `yield` usage in the other scheduler, although I'm sure I would get use to it.